documentation/docs/60-appendix/30-migrating-to-sveltekit-2.md

```diff
@@ -68,7 +68,7 @@ export function load({ fetch }) {
 
 ## goto(...) changes
 
-`goto(...)` no longer accepts external URLs. To navigate to an external URL, use `window.location = url`. The `state` object now determines `$page.state` and must adhere to the `App.PageState` interface, if declared. See [shallow routing](shallow-routing) for more details.
+`goto(...)` no longer accepts external URLs. To navigate to an external URL, use `window.location.href = url`. The `state` object now determines `$page.state` and must adhere to the `App.PageState` interface, if declared. See [shallow routing](shallow-routing) for more details.
```

Comment:
If you think about `url` as being the `URL` object then the previous code was correct. If we want to change this then we should probably also name the variable something else than `url`.

Reply:
`Location` and `URL` aren't the same thing, so the previous code _wasn't_ correct with that assumption (though it works just fine — this is an annoying typescript-tail-wagging-the-javascript-dog situation):

<img width="687" alt="image" src="https://github.com/sveltejs/kit/assets/1162160/b35ecf9f-aba0-4b90-9ac5-63f7baf92837">

We can't change the name of `url` without the `goto` signature being weird — there's really nothing that makes sense [here](https://kit.svelte.dev/docs/modules#$app-navigation-goto) _other_ than `url`, given its type:

```ts
function goto(url: string | URL, opts?: {...}): Promise<void>;
```

Reply:
That seems way less helpful than showing me code that works, even if it has a type error in the rare case of someone using a `URL` object rather than a string

Reply:
So you're suggesting to just accept this PR as-is then?

Reply:
yep. just slapped a 👍  on it